mkosi/kmod.py

```diff
@@ -218,12 +218,16 @@ def process_kernel_modules(
 
     with complete_step("Applying kernel module filters"):
         required = set(gen_required_kernel_modules(root, kver, include=include, exclude=exclude))
+        unneeded = set()
 
         with chdir(root):
             modules = sorted(modulesd.rglob("*.ko*"), reverse=True)
             firmware = sorted(firmwared.rglob("*"), reverse=True)
 
         for m in modules:
+            if m.is_relative_to(modulesd / "updates"):
+                continue
+
             if m in required:
```

Comment:
This is a bit shorter and the comment below, would probably make more sense here. 
```suggestion
            if m in required or m.is_relative_to(modulesd / "updates"):
```
That being said, if files under `updates/` are required, that logic should be in `gen_required_kernel_modules`.